public/data_source_connection/components/connections_bar.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { EuiPortal } from '@elastic/eui';
+import { combineLatest } from 'rxjs';
+import { DataPublicPluginStart, IDataPluginServices } from '../../../../../src/plugins/data/public';
+import { QueryEditorExtensionDependencies } from '../../../../../src/plugins/data/public/ui/query_editor/query_editor_extensions/query_editor_extension';
+import { useOpenSearchDashboards } from '../../../../../src/plugins/opensearch_dashboards_react/public';
+import { DataSourceSelector } from '../../../../../src/plugins/data_source_management/public';
+import { ConnectionsService } from '../services';
+
+interface ConnectionsProps {
+  dependencies: QueryEditorExtensionDependencies;
+  connectionsService: ConnectionsService;
+}
+
+export const ConnectionsBar: React.FC<ConnectionsProps> = ({ connectionsService }) => {
+  const { services } = useOpenSearchDashboards<IDataPluginServices>();
+  const {
+    savedObjects,
+    notifications: { toasts },
+  } = services;
+  const [isDataSourceEnabled, setIsDataSourceEnabled] = useState<boolean>(false);
+  const [uiService, setUiService] = useState<DataPublicPluginStart['ui'] | undefined>(undefined);
+  const containerRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    const subscription = combineLatest([
+      connectionsService.getUiService(),
+      connectionsService.getIsDataSourceEnabled$(),
+    ]).subscribe(([service, enabled]) => {
+      setUiService(service);
+      setIsDataSourceEnabled(enabled);
+    });
+
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, [connectionsService]);
+
+  const setContainerRef = useCallback((uiContainerRef) => {
+    if (uiContainerRef && containerRef.current) {
+      uiContainerRef.append(containerRef.current);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (!uiService || !isDataSourceEnabled) return;
+    const subscriptions = uiService.dataSourceContainer$.subscribe((container) => {
+      if (container === null) return;
+      setContainerRef(container);
+    });
+
+    return () => {
+      subscriptions.unsubscribe();
+    };
+  }, [
+    uiService,
+    uiService?.dataSourceContainer$,
+    containerRef,
+    setContainerRef,
+    isDataSourceEnabled,
+  ]);
+
+  useEffect(() => {
+    const subscriptions = connectionsService.getSelectedConnection$().subscribe((connection) => {
+      connectionsService.setSelectedConnection$(connection);
+    });
+    return () => {
+      subscriptions.unsubscribe();
+    };
+  }, [connectionsService]);
```

Comment:
why is this getting connection and setting the same connection?

Reply:
good catch. whenever Discover page just incase the client switched off page i can still set it in the service. but tbh it might just be left over. will look again post merge

Reply:
didn't fully understand.. but i just tested this PR and this section is giving me `Uncaught RangeError: Maximum call stack size exceeded`